providers.go

```diff
@@ -3,23 +3,26 @@ package configstore
 import (
 	"fmt"
 	"io/ioutil"
+	"log"
 	"os"
 	"path/filepath"
 	"strings"
 	"sync"
 	"time"
-	"unicode"
-	"unicode/utf8"
 
 	"github.com/ghodss/yaml"
-	"github.com/sirupsen/logrus"
 )
 
+// Logs functions can be overriden
+var LogErrorFunc = log.Printf
+var LogInfoFunc = log.Printf
+
 /*
 ** DEFAULT PROVIDERS IMPLEMENTATION
  */
 
 func errorProvider(s *Store, name string, err error) {
+	LogErrorFunc("error: %v", err)
```

Comment:
As variable is overridable by anybody, I would protect this path of code
```suggestion
	if LogErrorFunc != nil {
		LogErrorFunc("error: %v", err)
	}
```

Reply:
done